src/Compilers/Core/MSBuildTask/GenerateMSBuildAnalyzerConfig.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.CodeAnalysis.BuildTasks
+{
+    public class GenerateMSBuildAnalyzerConfig : Task
+    {
+        [Output]
+        public string ConfigFileContents { get; set; }
+
+        [Required]
+        public ITaskItem[] MetadataItems { get; set; }
+
+        [Required]
+        public ITaskItem[] PropertyItems { get; set; }
+
+        public GenerateMSBuildAnalyzerConfig()
+        {
+            ConfigFileContents = string.Empty;
+            MetadataItems = new ITaskItem[0];
+            PropertyItems = new ITaskItem[0];
+        }
+
+        public override bool Execute()
+        {
+            StringBuilder builder = new StringBuilder();
+
+            // we always generate global configs
+            builder.AppendLine("is_global = true");
+
+            // collect the properties into a global section
+            foreach (var prop in PropertyItems)
+            {
+                builder.Append("msbuild_property.")
+                       .Append(prop.ItemSpec)
```

Comment:
What does `ItemSpec` return in this context?

Reply:
The name of the actual property. ItemSpec is what's in the `Include` attribute. We include the properties on line 103 of `Microsoft.Managed.Core.targets` split out by their values. 

---
In reply to: [419752343](https://github.com/dotnet/roslyn/pull/43617#discussion_r419752343) [](ancestors = 419752343)